CMakeLists.txt

```diff
@@ -1,7 +1,9 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.12)
 project(dynamic_gazebo_models)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
```

Comment:
If you use this, I recommend also setting `CMAKE_CXX_EXTENSIONS`  to `OFF`, otherwise you'll get `gnu++17` instead of `c++17`.

Reply:
My instinct is to leave it as-is, if only to avoid overriding more defaults than strictly necessary, eg:

194 files: http://codesearch/?q=CMAKE_CXX_STANDARD
15 files: http://codesearch/?q=CMAKE_CXX_EXTENSIONS